feat: Add "Edit this page" button to docs

Calm-Rock opened the pull request:
This PR adds an "Edit" button on documentation pages. The button links directly to the GitHub edit page for the corresponding markdown file.

Fixes: https://github.com/SigNoz/signoz-web/issues/1756

Calm-Rock committed:
feat: Add "Edit this page" button to docs

This commit adds an "Edit this page" button next to the "Last updated" date on documentation pages.
The button links directly to the GitHub edit page for the corresponding markdown file.

- Modified  to include the edit link.
- Modified  to pass the edit link to .

Calm-Rock committed:
feat: Enhance "Edit this page" button in docs

This commit enhances the "Edit this page" button in the documentation section.
It now includes a pencil icon, is right-aligned, and matches the date's font color.

- Modified  to implement these changes.

Calm-Rock committed:
fix: Adjust "Edit this page" button color

This commit adjusts the color of the "Edit this page" button to strictly match the date's color,
removing any hover effects that might have caused a perceived color difference.

- Modified  to update the button's styling.

Calm-Rock committed:
Update DocContent.tsx

makeavish reviewed (commented):
Can you make button on prominent?

<img width="204" height="64" alt="Image" src="https://github.com/user-attachments/assets/5ff25e7c-bc4e-47fe-82e3-b80693cb70a9" />

Calm-Rock committed:
feat: Use Lucide React icon and enhance edit button styling

- Add Edit3 icon from lucide-react for consistency with codebase
- Make edit button more prominent with proper button styling
- Add hover effects and better visual hierarchy
- Improve accessibility with better contrast and spacing

Calm-Rock committed:
feat: Remove pencil icon and underline from edit button

- Remove Edit3 icon import and usage
- Add no-underline class to remove text decoration
- Keep the prominent button styling without the icon

Calm-Rock committed:
feat: Keep pencil icon but remove underline from edit button

- Restore Edit3 pencil icon from lucide-react
- Keep no-underline class to remove text decoration
- Maintain prominent button styling with icon and clean text

Calm-Rock committed:
feat: Use Edit icon to match the desired pencil icon style

- Change from Edit3 to Edit icon from lucide-react
- This matches the simple pencil icon style shown in the reference
- Maintain all other styling and functionality

Calm-Rock commented:
> Can you make button on prominent?
> 
> <img width="204" height="64" alt="Image" src="https://github.com/user-attachments/assets/5ff25e7c-bc4e-47fe-82e3-b80693cb70a9" />
> 

Done

Calm-Rock committed:
feat: Use existing Button component for consistency

- Replace custom styled anchor with Button component from @/components/ui/Button
- Use outline variant as suggested by front-end developer
- Automatically handles noopener noreferrer nofollow for SEO
- Improves component standardization across the web repo
- Maintains Edit icon and 'Edit on GitHub' text
